lib/rex/post/meterpreter/ui/console/command_dispatcher/stdapi/net.rb

```diff
@@ -385,70 +401,151 @@ def cmd_portfwd(*args)
 
     # Process the command
     case args.shift
-      when "list"
+      when 'list'
+
+        table = Rex::Ui::Text::Table.new(
+          'Header'    => 'Active Port Forwards',
+          'Indent'    => 3,
+          'SortIndex' => -1,
+          'Columns'   => ['Index', 'Local', 'Remote', 'Direction'])
 
         cnt = 0
 
         # Enumerate each TCP relay
         service.each_tcp_relay { |lhost, lport, rhost, rport, opts|
           next if (opts['MeterpreterRelay'] == nil)
 
-          print_line("#{cnt}: #{lhost}:#{lport} -> #{rhost}:#{rport}")
+          direction = 'Forward'
+          direction = 'Reverse' if opts['Reverse'] == true
+
+          if opts['Reverse'] == true
+            table << [cnt + 1, "#{rhost}:#{rport}", "#{lhost}:#{lport}", 'Reverse']
+          else
+            table << [cnt + 1, "#{lhost}:#{lport}", "#{rhost}:#{rport}", 'Forward']
+          end
 
           cnt += 1
         }
 
         print_line
-        print_line("#{cnt} total local port forwards.")
+        if cnt > 0
+          print_line(table.to_s)
+          print_line("#{cnt} total active port forwards.")
+        else
+          print_line('No port forwards are currently active.')
+        end
+        print_line
 
+      when 'add'
 
-      when "add"
+        if reverse
+          # Validate parameters
+          unless lport && lhost && rport
+            print_error('You must supply a local port, local host, and remote port.')
+            return
+          end
 
-        # Validate parameters
-        if (!lport or !rhost or !rport)
-          print_error("You must supply a local port, remote host, and remote port.")
-          return
-        end
+          channel = client.net.socket.create(
+            Rex::Socket::Parameters.new(
+              'LocalPort' => rport,
+              'Proto'     => 'tcp',
+              'Server'    => true
+            )
+          )
```

Comment:
I'm wondering if the `start_reverse_tcp_relay` function (below) should be responsible for the creation of the channel rather than it being created here and passed in. What do you people think? It feels like it's in the wrong spot here. Ping @bcook-r7 / @wvu-r7 .